### DISCUSSION TITLE
KoReader Sync

### DISCUSSION THREAD

**Discussion by Silther** [stumpapp/stump#509] [Question] [answered]
Does it work in both directions?
If you have multple E-Readers with KoReader, can you switch between them without losing progress? Or does it only upload your current progress?

**Comment by aaronleopold** [comment, marked as answer]
> Does it work in both directions?
> If you have multple E-Readers with KoReader, can you switch between them without losing progress? Or does it only upload your current progress? 

It should work in both directions, but it won't maintain different progress states per device.

KoReader sends some kind of identifier for the device being used, which Stump will attach to the record(s) it uses for tracking progress. Stump will update the device if it changes, but progress isn't loaded based off of the device requesting it. This means if another device asks for progress, it will send the progress from the other device (the record is shared). If the new device syncs progress, the device gets updated on that same record.

I'll add that devices are stored in the database when they are first encountered, but nothing is surfaced on the UI for that yet. I have thoughts/plans to allow you to add nicknames/aliases for them.
